src/main/java/net/snowflake/ingest/streaming/internal/DataValidationUtil.java

```diff
@@ -37,12 +45,26 @@ class DataValidationUtil {
   public static final int BYTES_8_MB = 8 * 1024 * 1024;
   public static final int BYTES_16_MB = 2 * BYTES_8_MB;
 
+  static final int MAX_VARIANT_LENGTH = BYTES_16_MB - 1024; // SNOW-664249
+
   private static final TimeZone DEFAULT_TIMEZONE =
       TimeZone.getTimeZone("America/Los_Angeles"); // default value of TIMEZONE system parameter
   private static final TimeZone GMT = TimeZone.getTimeZone("GMT");
 
   private static final ObjectMapper objectMapper = new ObjectMapper();
 
+  static {
```

Comment:
Could you add a comment explain what this is used for?

Reply:
Added a comment